sources/cfsignal/src/service_check.rs

```diff
@@ -0,0 +1,64 @@
+use crate::error::{self, Result};
+use log::trace;
+use snafu::ResultExt;
+use std::process::Command;
+
+#[derive(Debug, Clone, Eq, PartialEq, Ord, PartialOrd)]
+pub(crate) struct ServiceHealth {
+    /// Whether or not the service is healthy.
+    pub(crate) is_healthy: bool,
+    /// In the event of an unhealthy service, the service's exit code (if found).
+    pub(crate) exit_code: Option<i32>,
+}
+
+pub(crate) trait ServiceCheck {
+    fn system_running(&self) -> Result<ServiceHealth>;
+}
+
+pub(crate) struct SystemdCheck {}
+
+impl ServiceCheck for SystemdCheck {
+    fn system_running(&self) -> Result<ServiceHealth> {
+        if is_system_running()? {
+            return Ok(ServiceHealth {
+                is_healthy: true,
+                exit_code: None,
+            });
+        }
+        Ok(ServiceHealth {
+            is_healthy: false,
+            exit_code: Some(Ok(1)?),
+        })
+    }
+}
+
+struct Outcome {
+    exit: i32,
+    stdout: String,
+}
+
+impl Outcome {
+    fn is_exit_true(&self) -> bool {
+        self.exit == 0
+    }
+}
+
+fn systemctl(args: &[&str]) -> Result<Outcome> {
+    trace!("calling systemctl with '{:?}'", args);
+    let output = Command::new("systemctl")
+        .args(args)
+        .output()
+        .with_context(|| error::Command {
+            command: "systemctl",
+            args: args.iter().map(|&s| s.to_owned()).collect::<Vec<String>>(),
+        })?;
+    Ok(Outcome {
+        exit: output.status.code().unwrap_or(-1),
+        stdout: String::from_utf8_lossy(output.stdout.as_slice()).into(),
+    })
+}
+
+fn is_system_running() -> Result<bool> {
```

Comment:
Rather than returning `Result<bool>` here, it might be useful to return the exit code (`i32`?) so that it can be logged further up.  Using the `bool` here means that the `exit_code` field in `ServiceCheck` is always either `None` or `Some(Ok(1)?)`.